CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+
+2. You can either work on an existing issue or you have to create a descriptive issue before you can make code changes. 
+
+3. Create a branch from `master` for the issue you are working on. 
+
+   * Use this naming scheme: 
+     * `<issue-type>/<issue-number>-<short-description>`. 
+     * Issue types are 'enhancement', 'feature' and 'bugfix'.
+
+   * Example: 
+     * `enhancement/122-update-of-pom.xml-dependencies`
+
+4. If you intend to do fundamental changes, e.g. changing APIs, please contact us beforehand. This is to avoid breaking existing functionality.
+
+5. Before creating a pull request, ensure that
+   1. all tests are are passing,
+   2. the comment of the pull request contains at least this line: 
+      * `closes #<issue-number>` .
+
+6. At least two active maintainers of the projects need to review your code before it is allowed to be merged.
+
+
+
+#### How to Make a Bug Report
+
+* Add a descriptive summary and/or background information.
+* Tell us what behavior of the application you expected and what you actually encountered.
+* Screenshots and log files are very helpful for us.
+* You should be able to reproduce the bug and enable us to do the same. Write down every single step that leads to the bug.
+* If you have a guess why this bug might be happening or why you think it didn't work, then tell us.
+
+
+
+#### Recommended Terminology
+
+To avoid misunderstandings, we have decided to provide a uniform terminology recommendation for the most important concepts in CoatRack. Try to avoid terms like 'server' and 'client' because the Gateway can be both in different contexts and therefore leads to confusion. As this recommendation did not exist from the very beginning of the project, you may encounter different wordings for the same concept. Here is an overview:
+
+| Recommended Wording      | Also Used Wording              | Description                                                  |
+| ------------------------ | ------------------------------ | ------------------------------------------------------------ |
+| CoatRack Web Application | (CoatRack) Admin, ygg-admin    | This is the central CoatRack component which can be reached at [coatrack.eu](https://coatrack.eu/). It offers a web UI to define services as well as creating gateways and API keys. |
+| Gateway                  | Proxy, ygg-proxy               | It grants access to the specific services if a Service User sends a valid API key to the gateway. |
+| Service Provider (\*)        | Service Offerer, Service Admin | This person defines services as well as API keys which grant access to those services. The Service Provider also hosts the gateway that provides the services. |
+| Service User (\*)             | Service Consumer               | This person can access services by contacting a gateway and providing a valid API key. |
+
+(\*) A real person can be a Service User and a Service Provider at the same time.
+
+
+
+## Contributor License Agreement
```

Comment:
I would suggest to move this part further to the top, directly after the introduction, before explaining all the workflow stuff etc.

You have actually put a note there "Note: If you want to contribute to CoatRack, ..."

That is where I would put this section.